app/models/mixins/dialog_option_mixin.rb

```diff
@@ -0,0 +1,34 @@
+module DialogOptionMixin
+  def self.get_dialog_option(key, value, from)
+    # Return value - Support array and non-array types
+    data = value.nil? ? from[key] : value
+    data.kind_of?(Array) ? data.first : data
+  end
+
+  def get_dialog_option(key, value = nil)
+    DialogOptionMixin.get_dialog_option(dialog_key(key), value, dialog_options)
+  end
+
+  def get_dialog_option_decrypted(key, value = nil)
+    raise ArgumentError, "#{key} cannot be decrypted" unless dialog_option_encrypted?(key)
```

Comment:
Would it be useful to also have a`try_decrypted ` similar to what MiqPassword has?
https://github.com/ManageIQ/manageiq-password/blob/master/lib/manageiq/password.rb#L96

Reply:
@gmcculloug 
For the service dialog the field type is appended to the key name e.g.
password::dialog_ldap_password = {v2}......

This is checking if the key has a password:: prefix.

Reply:
I think the question is more whether this PR should create a `dialog_option_try_decrypt` that works similar to MiqPassword's `try_decrypt`, thus making it easier for callers to not hit these nuances.

Reply:
@Fryguy from automate method perspective we have had the users always specify which attribute they want to be decrypted

https://github.com/ManageIQ/manageiq-automation_engine/blob/445f6a0505eb025638853be5a59f41832cb1fa85/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_object_common.rb#L28

From their use case they need to decrypt a password just before they make a connection to a remote server, they would not be going thru a loop and trying to decrypt every attribute or option key. They would be either using a specific attribute name or a specific key name at the point where they need to use it.